test/system_test.rb

```diff
@@ -0,0 +1,290 @@
+require_relative 'test_helper'
+
+describe "System class" do
+  describe "System instantiation" do
+    before do
+      @sys = Hotel::System.new
+    end
+    
+    it "is an instance of System" do
+      expect(@sys).must_be_instance_of Hotel::System
+    end
+    
+    it "is set up for specific attributes and data types" do
+      [:rooms, :reservations, :dates, :hotelblocks].each do |prop|
+        expect(@sys).must_respond_to prop
+      end
+      
+      expect(@sys.rooms).must_be_instance_of Array
+      expect(@sys.rooms.first).must_be_instance_of Hotel::Room
+      expect(@sys.rooms.last).must_be_instance_of Hotel::Room
+      expect(@sys.reservations).must_be_instance_of Array
+      expect(@sys.dates).must_be_instance_of Array
+      expect(@sys.hotelblocks).must_be_instance_of Hash
+    end
+    
+    it "allows access to the list of all rooms" do
+      expect(@sys.rooms.length).must_equal 20
+      expect(@sys.rooms[11].id).must_equal 12
+      expect(@sys.rooms[11].cost).must_equal 200.0
+    end
+  end
+  
+  describe "room reservation creation" do
+    before do
+      @sys = Hotel::System.new
+      @first_res = @sys.make_reservation(::Date.parse('2019-06-05'), ::Date.parse('2019-06-08'))
+    end
+    
+    it "generates reserves a room for a given date range" do
+      expect(@first_res).must_be_instance_of Hotel::Reservation
+      
+      expect(@first_res.id).must_equal 1
+      expect(@first_res.start_date).must_equal ::Date.parse('2019-06-05')
```

Comment:
Consider assigning dates to variables in your tests to make it clear what you are testing.
```suggestion
      start_date = Date.parse('2019-06-05')
      end_date = start_date + 3
```